### ISSUE TITLE
[BUG] [JAVA] dateLibrary java8 does not work with xml

### ISSUE THREAD

**Issue by davidmoten** [OpenAPITools/openapi-generator#8134]
##### Description

*openapi-generator-maven-plugin* generates java code with jaxb annotations when `withXml=true`. If I set `dateLibrary=java8` then datetime fields get mapped to `OffsetDateTime` types in the generated code. However, when I go to use the generated code the xml returned from an openapi service cannot be deserialized into java objects because no `XmlTypeAdapter` annotation has been defined on the `OffsetDateTime` field and it is required so that jaxb can deserialize a string datetime into that field. 

My workaround is to use `dateLibrary=legacy` and generate `java.util.Date` types instead. I would like to use `java8` types.

##### openapi-generator version

4.3.1 (I have not tested with 5.0.0beta3)

##### Steps to reproduce

* create an openapi definition file with datetime field(s) in the returned data from a service call. 
* Generate java code using *openapi-generator-maven-plugin*, `withXml=true`. 
* Check that there is an `XmlTypeAdapter` annotation on the datetime field in the java code.

##### Suggest a fix

Generate an `XmlTypeAdapter` for `OffsetDateTime` (and other new time classes) or include the adapters in a runtime library. Then annotate the `OffsetDateTime` fields in generated code with the appropriate `XmlTypeAdapter`.

**Comment by wing328** [comment]
Please give it a try with 5.0.0-beta3 or the latest master as there's been lots of enhancements, bug fixes since v4.3.1

**Comment by davidmoten** [comment]
Doesn't work with 5.0.0-beta3. This is the field declaration for a datetime:

```java
  public static final String JSON_PROPERTY_TIME = "time";
  @XmlAttribute(name = "time")
  private OffsetDateTime time;
```
